Introduce roles for files related to calculations (#523)
Materials-Consortia/OPTIMADE: optimade.rst

--------------------

The :entry:`calculations` entries have the properties described above in section `Properties Used by Multiple Entry Types`_.
Related :entry:`files` entries specified as relationships (as described in section `Relationships`_) list files used or produced by a calculation.
They MAY be defined as either input or output files by the :field:`role` field inside the :field:`meta` dictionary of the JSON:API resource identifier object.
The allowed values are :val:`input` for input files and :val:`output` for output files.

> Absence of :field:`role` or :val:`null` means that the role is undefined and SHOULD NOT be assumed.

Reviewer: ```suggestion
If the :field:`role` field is not provided or the value is set to :val:`null`, a default role SHOULD NOT be assumed.
```

> It is RECOMMENDED to order the output files in the list of calculation's relationships in a way to prioritize the files most likely to contain details which could be used to automatically analyze the performed calculation.

Reviewer: ```suggestion
It is RECOMMENDED to order the output files in the list of calculation's relationships in a way that prioritizes the files most likely to contain details which could be used to automatically analyze the performed calculation.
```

> This way output parsing software would need to read in a small number of calculation output files to possibly identify the code, type and possibly parameters of calculation performed.

Reviewer: ```suggestion
Such an approach minimizes the number of calculation output files that need to be analyzed to possibly identify the code, type and parameters of the performed calculation.
```
